iothub/device/src/Authentication/IotHubConnectionStringBuilder.cs

```diff
@@ -57,74 +57,50 @@ internal IotHubConnectionStringBuilder()
         }
 
         /// <summary>
-        /// Creates a connection string based on the hostname of the IoT hub and the authentication method passed as a parameter.
-        /// </summary>
-        /// <param name="hostname">The fully-qualified DNS hostname of IoT hub</param>
-        /// <param name="authenticationMethod">The authentication method that is used</param>
-        /// <returns>A new instance of the <see cref="IotHubConnectionStringBuilder"/> class with a populated connection string.</returns>
-        public static IotHubConnectionStringBuilder Create(string hostname, IAuthenticationMethod authenticationMethod)
-        {
-            return Create(hostname, null, authenticationMethod);
-        }
-
-        /// <summary>
-        /// Creates a connection string based on the hostname of the IoT hub, the hostname of Gateway and the authentication
+        /// Creates a connection string, based on the hostname of the IoT hub, the hostname of Gateway and the authentication
         /// method passed as a parameter.
         /// </summary>
-        /// <param name="hostname">The fully-qualified DNS hostname of IoT hub</param>
-        /// <param name="gatewayHostname">The fully-qualified DNS hostname of the gateway</param>
-        /// <param name="authenticationMethod">The authentication method that is used</param>
+        /// <param name="authenticationMethod">The authentication method that is used.</param>
+        /// <param name="hostName">The fully-qualified DNS host name of IoT hub.</param>
+        /// <param name="gatewayHostName">The fully-qualified DNS host name of the gateway.</param>
         /// <returns>A new instance of the <see cref="IotHubConnectionStringBuilder"/> class with a populated connection string.</returns>
-        public static IotHubConnectionStringBuilder Create(string hostname, string gatewayHostname, IAuthenticationMethod authenticationMethod)
+        public IotHubConnectionStringBuilder(IAuthenticationMethod authenticationMethod, string hostName, string gatewayHostName = null)
```

Comment:
Do you think this would be a useful helper method for users or could we make this internal?

Reply:
Helper = IotHubConnectionStringBuilder itself

Reply:
I think we could make it internal and only bring it back if a customer makes a case for why it was valuable to them.

Reply:
Well, no because IAuthenticationMethod refers to this type in the impl. So it wouldn't be that easy.